src/poll_manager.ts

```diff
@@ -0,0 +1,149 @@
+import type { StreamChat } from './client';
+import type {
+  CreatePollData,
+  DefaultGenerics,
+  ExtendableGenerics,
+  PollResponse,
+  PollSort,
+  QueryPollsFilters,
+  QueryPollsOptions,
+} from './types';
+import { Poll } from './poll';
+import { FormatMessageResponse } from './types';
+import { formatMessage } from './utils';
+
+export class PollManager<SCG extends ExtendableGenerics = DefaultGenerics> {
+  private client: StreamChat<SCG>;
+  private pollCache = new Map<string, Poll<SCG>>();
+  private unsubscribeFunctions: Set<() => void> = new Set();
+
+  constructor({ client }: { client: StreamChat<SCG> }) {
+    this.client = client;
+  }
+
+  get data(): Map<string, Poll<SCG>> {
+    return this.pollCache;
+  }
+
+  public fromState = (id: string) => {
+    return this.pollCache.get(id);
+  };
+
+  public registerSubscriptions = () => {
+    if (this.unsubscribeFunctions.size) {
+      // Already listening for events and changes
+      return;
+    }
+
+    this.unsubscribeFunctions.add(this.subscribeMessageNew());
+    this.unsubscribeFunctions.add(this.subscribePollUpdated());
+    this.unsubscribeFunctions.add(this.subscribePollClosed());
+    this.unsubscribeFunctions.add(this.subscribeVoteCasted());
+    this.unsubscribeFunctions.add(this.subscribeVoteChanged());
+    this.unsubscribeFunctions.add(this.subscribeVoteRemoved());
+  };
+
+  public unregisterSubscriptions = () => {
+    this.unsubscribeFunctions.forEach((cleanupFunction) => cleanupFunction());
+    this.unsubscribeFunctions.clear();
+  };
+
+  public createPoll = async (poll: CreatePollData<SCG>) => {
+    const { poll: createdPoll } = await this.client.createPoll(poll);
+
+    return new Poll({ client: this.client, poll: createdPoll });
```

Comment:
Why is the newly created poll not being cached?

Reply:
It's not since presence in the cache is directly tied to the poll actually being sent as a message somewhere (i.e appearing when we `queryChannels`, or a `message.new` event arrives with a `poll` inside of it etc). But it might be a good point to perhaps not even create a poll instance, since in this case we're really only interested in the ID of the poll that got created.